### DISCUSSION TITLE
Database continues to restart due to "Database export failed the last time. Most likely was the export time not high enough."

### DISCUSSION THREAD

**Discussion by wildernessfamily** [nextcloud/all-in-one#4916] [Questions] [answered]
### Steps to reproduce
1. NextCloud AIO was up and running with no problems. I was running semi-low on storage space and decided to move it over onto a large storage drive. Doing so, I needed to change the `datadir` location. Followed the procedures stated at `https://github.com/nextcloud/all-in-one#how-to-change-the-default-location-of-nextclouds-datadir`
2. Stopped NextCloud-AIO. Removed the `nextcloud-aio-mastercontainer`. Stopped Docker. Performed a `rsync -av` from old location to new location. Updated my Docker Compose YAML file by changing the `NEXTCLOUD_DATADIR` to the new location. Started Docker. And Performed the normal `docker compose up -d && docker compose logs -f`
3. The database container would continue to restart. The nextcloud-aio-database container log would show a constant repeat of the following:
```text
Database export failed the last time. Most likely was the export time not high enough.
Please report this to https://github.com/nextcloud/all-in-one/issues. Thanks!
```
4. I tried manually removing each continuer `nextcloud-aio-...`. I tried manually removing each image allowing AIO to repull the image using `docker image rm image-id`.
5. I can't get past the repeating restart of the database.
6. I did creat a docker container with PostGres15 and Adminer and attach it to the `nextcloud-aio-database` volume. I was able to verify the database is intact. I also exported the database to a .sql file, just in case. I also made a copy of the volume with the `cp -r /var/lib/docker/volume/nextcloud-aio-database/_data /backup-location/`

### Expected behavior 
Normal startup behavior.

### Actual behavior 
See above.

### Host OS 
Raspberry Pi 4, 8GB. Ubuntu 22.04.4 LTS (Jammy) | Linux 5.15.0-1055-raspi| Arm64

#### Nextcloud AIO version 
v9.0.1

#### Current channel 
Latest

**Comment by szaimen** [comment]
Hi, I think the export failed because you did not stop the containers via the AIO interface. 

Can you restore a backup from before the changes that you did and check if that fixes the problem?

**Comment by wildernessfamily** [comment]
Hi, normally I respond within a few hours. My house is under construction & I'm staying at a hotel during construction. Things are a bit hectic right now. My apologies for not responding quicker. 
I tried to make a backup, but I didn't have enough storage space. That's one of many reasons I decided to go from 4tb to 16tb. I was crossing my fingers that this would go without any problems. That said, I do not for sure, without any doubt, that I did stop all the containers in the AIO interface. I did not do it at the command prompt. I received a notification that the master container had an update. I wanted to apply the update first, prior to the move. I used the button to stop all containers in the AIO Admin interface.

**Reply by szaimen** [reply]
How big is your database? Did it actually take more than 30min to shut down the containers?

**Reply by wildernessfamily** [reply]
Oh..., no, not at all. Normally, it takes just a few minutes to shut down normally. Less than 5 minutes to shut down all the containers. I received a notification that there was an update for the master container. I thought let's just update all the containers before I move over to the new 16TB drive. I went into the AIO Admin, click to shut down the containers. At some point, the browser page timeout. I went to the CLI, and saw the database was stalled. This was about 30-40 mins into the shutdown. Finally, I just did the normal `docker stop nextcloud-aio-database`, which worked. The container stopped within seconds with no problems, But, then the problem started.
I tried and tried to fix this. I finally, had to reach out for your help. Doing so, I thought it be best to attach a Postgres 15 container to the database volume. I exported the database to an SQL file. It's exactly 395mb. I did check the SQL. It's intact.
FYI, thank you for the time you are putting in to assist me. It's greatly appreciated!
<img width="673" alt="Screen Shot 2024-07-12 at 19 49 43" src="https://github.com/user-attachments/assets/6311a111-36cd-4dd9-8675-881daf618e2a">

**Reply by szaimen** [reply]
> docker stop nextcloud-aio-database

This is the problem. It does not respect the timeout and thus the export failed because this command only has a timeout of 10s

**Comment by wildernessfamily** [comment]
Okay, so, I'm not sure what that means for me to fix the problem?

**Comment by szaimen** [comment]
Do you have a backup from before the incident happened?

**Reply by szaimen** [reply, marked as answer]
Sorry, I just remembered that you don't have a backup... I fear this is a unlucky situation. Maybe someone in the forum has some ideas how to recover from this situation: https://help.nextcloud.com. In worst case you need to start with a fresh instance and import the files: https://github.com/nextcloud/all-in-one/blob/main/migration.md#migrate-only-the-files

**Comment by wildernessfamily** [comment]
Hi, again, thank you for taking the time to respond and help with my problem. I was pretty sure that was going to happen. The only thing that concerns me is the database. I have a copy of the database as a SQL file as I exported just in case this did happen. I use Deck, Recipes, Passwords, OTA, GPX for my Garmin GPS, etc., plugins. I've done multiple migrations per the doc for testing. Scanning the file is easy. I created a script for NFS to scan the files after transfer. I use the app nextcloud-inotifyscan. I'm very familiar with all that. The one thing, which might be the most important, is the database. I can't lose the plugin data that's in the DB, for the plugins I previously mentioned. All the other data, I can easily re-create. Re-creating the users, re-installing the plugins, setting up colors, etc. A couple of hours, and I'll have that all back to how it was previously. After, I do a fresh install, can I stop all the containers, and then attach to the Postgres database container, remove all the tables, and then import my SQL file? Do you know if that would work? Have you done that before? Another question is, if that does not work, is it possible to import the tables just for those plugins (the tables associated with the plugins I use, like passwords and OTA), install the plugins and then hopefully🤞  plugins will be able to recognize and use the imported tables. Any guidance on this would GREATLY be appreciated 😃🙏. Again, I can't express how appreciative I am that you are taking time from your daily life to assist me. My family and I appreciate your help! Thank you!

**Reply by szaimen** [reply]
> I have a copy of the database as a SQL file as I exported just in case this did happen

is this a dump file? If so, you should be able to place it inside the `/var/lib/docker/volumes/nextcloud_aio_database_dump/_data/` location with the name `database-dump.sql`, remove `export.failed` file in there and try to restart the database container which might allow it to start correcty

**Comment by wildernessfamily** [comment]
Oh..., I was not aware of that. Yes, it's a complete dump file. I'm going to give that a try this evening. Thanks for that info! Just in case anyone else has this issue. Don't remove the volume just because your database refuses to start. I pulled just the normal Postgres container from Docker Hub with the version of the current database, which was in my case was version 15. I also added Adminer as I wanted to inspect the tables to make sure the entire database was intact. I attached the database volume to the Postgres container. I was able to log into Adminer, and dumped the entire SQL database into a SQL file. Of course, you can leave off Adminer, and just log into the Postgres container and run the dump command. But, as I stated, I wanted to examine the tables visibly, prior to dumping the database. I wasn't sure if the database was intact. I'll report back on placing the sql file in `/var/lib/docker/volumes/nextcloud_aio_database_dump/_data/` naming the SQL file `database-dump.sql` and making sure to remove `export.failed` and trying a restart. 🤞:) Thank you again for your assistance @szaimen
